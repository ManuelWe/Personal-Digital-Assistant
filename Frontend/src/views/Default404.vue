<template>
  <div
    id="Default404"
    class="hero-body"
  >
    <div class="container">
      <h1 class="is-size-3">
        Sorry. This page doesnt exist.
      </h1>
      <router-link
        class="router-link"
        :to="{name: 'landingPage'}"
      >
        <div class="button">
          Back to Home
        </div>
      </router-link>
    </div>
  </div>
</template>

<script>
export default {
  data() {
    return {
    };
  },
};
</script>
