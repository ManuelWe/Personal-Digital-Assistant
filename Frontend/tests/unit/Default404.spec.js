import { createLocalVue, mount } from '@vue/test-utils';
import Buefy from 'buefy';
import VueRouter from 'vue-router';
import Default404 from '@/views/Default404.vue';

const localVue = createLocalVue();
localVue.use(Buefy);
localVue.use(VueRouter);

const routes = [
  {
    path: '/default',
    name: 'landingPage',
    component: Default404,
  },
];
const router = new VueRouter({
  routes,
});

function factory() {
  return mount(Default404, {
    localVue,
    router,
  });
}

describe('Default404.vue', () => {
  describe('Elements are rendered correctly', () => {
    it('Component renders', () => {
      const wrapper = factory();
      expect(wrapper).toMatchSnapshot();
    });

    it('Link to home page is displayed', () => {
      const wrapper = factory();
      expect(wrapper.find('.router-link').exists()).toBe(true);
    });
  });
});
