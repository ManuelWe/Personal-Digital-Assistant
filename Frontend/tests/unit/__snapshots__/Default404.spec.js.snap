// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default404.vue Elements are rendered correctly Component renders 1`] = `
<div id="Default404" class="hero-body">
  <div class="container">
    <h1 class="is-size-3">
      Sorry. This page doesnt exist.
    </h1> <a href="#/default" class="router-link">
      <div class="button">
        Back to Home
      </div>
    </a>
  </div>
</div>
`;
